dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/SwordConfigurationFactory.java

```diff
@@ -66,35 +51,13 @@ public SwordConfiguration createSwordConfiguration() {
      * @return temporary Sword directory, based on system config
      */
     private String createTempDirectory() {
-        String tmpFileDir = systemConfig.getFilesDirectoryProperty();
-        if (tmpFileDir != null) {
-            String swordDirString = tmpFileDir + File.separator + "sword";
-            File swordDirFile = new File(swordDirString);
-            /**
-             * @todo Do we really need this check? It seems like we do because
-             * if you create a dataset via the native API and then later try to
-             * upload a file via SWORD, the directory defined by
-             * dataverse.files.directory may not exist and we get errors deep in
-             * the SWORD library code. Could maybe use a try catch in the doPost
-             * method of our SWORDv2MediaResourceServlet.
-             */
-            if (swordDirFile.exists()) {
-                return swordDirString;
-            } else {
-                boolean mkdirSuccess = swordDirFile.mkdirs();
-                if (mkdirSuccess) {
-                    return swordDirString;
-                } else {
-                    String msgForSwordUsers = ("Could not determine or create SWORD temp directory. Check logs for details.");
-                    // sadly, must throw RunTimeException to communicate with SWORD user
-                    throw new RuntimeException(msgForSwordUsers);
-                }
-            }
-        } else {
-            String msgForSwordUsers = ("JVM option \"" + SystemConfig.FILES_DIRECTORY + "\" not defined. Check logs for details.");
-            // sadly, must throw RunTimeException to communicate with SWORD user
-            throw new RuntimeException(msgForSwordUsers);
+        String tmpFileDir = systemConfig.getFilesDirectory();
+        File swordDirFile = new File(tmpFileDir + File.separator + "sword");
```

Comment:
This would work but is not exactly what I had in mind. I meant:
```
File swordDirFile = new File(tmpFileDir, "sword");
```
But I'm not insisting on this very much

Reply:
Ok, File.separator doesn't make any difference here so it's cleaner to omit it and use (String,String) constructor.